crates/network/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+        U256,
+    },
+    thiserror::Error,
+};
+
+/// Represents each available network
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Network {
+    Mainnet = 1,
+    Goerli = 5,
+    Gnosis = 100,
+    Sepolia = 11155111,
+    ArbitrumOne = 42161,
+    Base = 8453,
+}
+
+impl Network {
+    /// Returns the network's chain ID
+    pub fn chain_id(&self) -> u64 {
+        match self {
+            Self::Mainnet => 1,
+            Self::Goerli => 5,
+            Self::Gnosis => 100,
+            Self::Sepolia => 11155111,
+            Self::ArbitrumOne => 42161,
+            Self::Base => 8453,
+        }
+    }
+
+    /// Returns the canonical name of the network on CoW Protocol.
+    pub fn name(&self) -> &'static str {
+        // You can find a list of available networks by network and chain id here:
+        // https://chainid.network/chains.json
+        match &self {
+            Self::Mainnet => "Ethereum / Mainnet",
+            Self::Goerli => "Ethereum / Goerli",
+            Self::Gnosis => "xDAI",
+            Self::Sepolia => "Ethereum / Sepolia",
+            Self::ArbitrumOne => "Arbitrum One",
+            Self::Base => "Base",
+        }
+    }
+
+    pub fn default_amount_to_estimate_native_prices_with(&self) -> Option<U256> {
+        match &self {
+            Self::Mainnet | Self::Goerli | Self::Sepolia | Self::ArbitrumOne | Self::Base => {
+                Some(10u128.pow(18).into())
+            }
+            Self::Gnosis => Some(10u128.pow(21).into()),
+        }
+    }
+
+    /// Returns the block time in milliseconds
+    pub fn block_time_in_ms(&self) -> u64 {
+        match self {
+            Self::Mainnet => 13_000,
+            Self::Goerli => 13_000,
+            Self::Gnosis => 5_000,
+            Self::Sepolia => 13_000,
+            Self::ArbitrumOne => 250,
+            Self::Base => 2_000,
+        }
+    }
+
+    /// Returns the number of blocks that fits into the given time (in
+    /// milliseconds)
+    pub fn number_of_blocks_in(&self, time_in_ms: u64) -> f64 {
+        time_in_ms as f64 / self.block_time_in_ms() as f64
+    }
+}
+
+impl TryFrom<u64> for Network {
+    type Error = Error;
+
+    /// Initializes `Network` from a chain ID, returns error if the chain id is
+    /// not supported
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        match value {
+            1 => Ok(Self::Mainnet),
+            5 => Ok(Self::Goerli),
+            100 => Ok(Self::Gnosis),
+            11155111 => Ok(Self::Sepolia),
+            42161 => Ok(Self::ArbitrumOne),
+            8453 => Ok(Self::Base),
+            _ => Err(Error::ChainIdNotSupported(value)),
+        }
+    }
+}
+
+impl TryFrom<U256> for Network {
+    type Error = Error;
+
+    /// Initializes `Network` from a chain ID, returns error if the chain id is
+    /// not supported
+    fn try_from(value: U256) -> Result<Self, Self::Error> {
+        value.as_u64().try_into()
```

Comment:
Since we already have error handling in this function we might as well return the appropriate error if `value > u64::MAX.into()` instead of truncating the value to make it fit into `u64`.
Doesn't really matter in practice but seems like the right thing to do.